e3db/client.py

```diff
@@ -1409,3 +1409,55 @@ def read_file(self, record_id, destination_filename):
             version=response_json['meta']['version'],
             plain=response_json['meta']['plain']
         )
+
+    def write_note(self, data: dict, recipient_encryption_key: str, recipient_signing_key: str, options: NoteOptions):
+        """
+        Public Method to make a note, encrypt it locally, and send it
+        to the server.
+
+        Parameters
+        ----------
+        data : dict 
+
+
+        recipient_encryption_key : str
+
+
+        recipient_signing_key : str
+
+
+        options : types.NoteOptions
+
+
+        Returns
+        -------
+        e3db.Note
+            Decrypted E3DB note
+        """
+        # Do we need a KeyPair object? 
+        # I think it makes it cleaner, but would agree if it was determined that it 
+        # did not need to be its own type. 
+        signing_keys = KeyPair(self.public_signing_key, self.private_signing_key)
+
+        encryption_keys = KeyPair(self.public_key, self.private_key)
+
+        return self.__write_note(data, options, recipient_signing_key, signing_keys,
+                                recipient_encryption_key, encryption_keys)
+
+    def __write_note(self, data, options: NoteOptions, recipient_signing_key, signing_keys: KeyPair,
+                     recipient_encryption_key, encryption_keys: KeyPair):      
+        """
+        Private Method to create and encrypt a note object, fetch TSV1 authentication,
+        and return the signed note. 
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        """
+        url = self.__get_url("v2", "storage", "notes")
+        note_keys = NoteKeys(None, recipient_signing_key, signing_keys.public_key, encryption_keys.public_key, None)
+        note = Note(data, note_keys, options)
```

Comment:
Just leaving a comment here that write_note is not yet good for production as it does not encrypt and sign the data

Reply:
Yup, I should have been more explicit about that in my top-level comment

Reply:
Does pip (by default) install from the latest commit or from a tagged version?

Reply:
pip installs from a tag that gets pushed to pypi, not related to github.

Reply:
This is not a blocker to get it merged into the feature branch. Just a note for reviewers and future us when seeing what needs to happen